rdt/transformers/boolean.py

```diff
@@ -85,4 +85,4 @@ def reverse_transform(self, data):
 
             data = pd.Series(data)
 
-        return np.round(data).astype('boolean').astype('object')
+        return np.round(data).clip(0, 1).astype('boolean').astype('object')
```

Comment:
Why do we cast as 'boolean' and then as 'object'? Is there a reason to not just cast as the latter?

Reply:
We have to first cast as `boolean` to convert the rounded floats to boolean values. Then it's cast to `object`, since that's the expected return type (a series with both `bool` and null types).